src/src/com/microsoft/aad/adal/DefaultTokenCacheStore.java

```diff
@@ -235,7 +235,7 @@ public void removeAll() {
             }
         }
 
-        return tokens.iterator();
+        return tokens;
```

Comment:
The list is created of a fixed size
ArrayList<TokenCacheItem> tokens = new ArrayList<TokenCacheItem>(results.values().size());

Addition is conditional
if (decryptedValue != null)

Will null be returned for any item not set?